src/tests/TouchFree.test.ts

```diff
@@ -144,13 +165,139 @@ describe('TouchFree', () => {
+
+            expect(addedEvents).toEqual(listeners);
+        });
+
+        it('should handle registering duplicate events', () => {
+            TouchFree.RegisterAnalyticEvents(['pointerdown']);
+            addedEvents.push('pointerdown');
+            expect(TouchFree.GetRegisteredAnalyticEventKeys()).toEqual(addedEvents);
+            TouchFree.RegisterAnalyticEvents(['pointerdown']);
+            expect(TouchFree.GetRegisteredAnalyticEventKeys()).toEqual(addedEvents);
+            TouchFree.RegisterAnalyticEvents(['keypress', 'keypress']);
+            addedEvents.push('keypress');
+            expect(TouchFree.GetRegisteredAnalyticEventKeys()).toEqual(addedEvents);
+
+            expect(addedEvents).toEqual(listeners);
+        });
+    });
+
+    describe('UnregisterAnalyticEvents', () => {
+        const listeners: string[] = [];
+        const defaults: AnalyticEventKey[] = ['touchstart', 'touchmove', 'touchend'];
+
+        const mock = jest.spyOn(document, 'removeEventListener').mockImplementation((event, _callback, option) => {
+            listeners.push(event);
+            expect(option).toBeTruthy();
+        });
+        afterAll(mock.mockRestore);
+
+        beforeEach(() => {
+            TouchFree.UnregisterAnalyticEvents();
+            TouchFree.RegisterAnalyticEvents(defaults);
+            expect(TouchFree.GetRegisteredAnalyticEventKeys()).toEqual(defaults);
+            listeners.length = 0;
+        });
+
+        it('should unregister specified events', () => {
+            TouchFree.UnregisterAnalyticEvents(['touchstart', 'touchmove']);
+            expect(TouchFree.GetRegisteredAnalyticEventKeys()).toEqual(['touchend']);
+            TouchFree.UnregisterAnalyticEvents(['touchend']);
+            expect(TouchFree.GetRegisteredAnalyticEventKeys()).toEqual([]);
+
+            expect(listeners).toEqual(defaults);
+        });
+
+        it('should unregister all events if none specified', () => {
+            TouchFree.UnregisterAnalyticEvents();
+            expect(TouchFree.GetRegisteredAnalyticEventKeys()).toEqual([]);
+            expect(listeners).toEqual(defaults);
+        });
+
+        it('should handle un-registering non registered events', () => {
+            expect(TouchFree.GetRegisteredAnalyticEventKeys()).not.toContain('pointerdown');
+            TouchFree.UnregisterAnalyticEvents(['pointerdown']);
+            expect(TouchFree.GetRegisteredAnalyticEventKeys()).toEqual(defaults);
+
+            expect(listeners).toEqual([]);
+        });
+    });
+
+    describe('AnalyticSessionEvents', () => {
+        beforeAll(() => {
+            TouchFree.RegisterAnalyticEvents(['pointerdown', 'keypress']);
+        });
+        it('should increment AnalyticSessionEvents count when event is triggered', () => {
+            TouchFree.RegisterAnalyticEvents(['pointerdown', 'keypress']);
+            document.dispatchEvent(new Event('pointerdown'));
```

Comment:
:shrug: I think we could probably do all these `new Event` calls once for each type rather than create a new one each time. Not that it'd save much performance, but it'd save a bit